packages/flutter_tools/test/general.shard/android/adb_log_reader_test.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/android/android_device.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+const int kLollipopVersionCode = 21;
+const String kLastLogcatTimestamp = '11-27 15:39:04.506';
+
+void main() {
+  testWithoutContext('AdbLogReader calls adb logcat with expected flags apiVersion 21', () async {
```

Comment:
minor nit: `group`?

Reply:
I'd rather leave it out if we wouldn't have more than one, since it introduces more rightward drift